src/vimeoplayer/views/Vimeoplayer.js

```diff
@@ -0,0 +1,472 @@
+
+			// Initialize debug counter
+			if (!this._debugCount) {
+				this._debugCount = 0;
+			}
+
+			// Log
+			console.log.apply(
+				console,
+				[this.getVideoId(), this._debugCount++].concat(
+					Array.prototype.slice.call(arguments)
+				)
+			);
+		}
+	}
+
+	/**
+     * Return true or false if the player is already initialized
+     *
+     * @private
+     * @return {boolean}
+     */
+	_hasPlayer() {
+		return !!this._player;
+	}
+
+	/**
+     * Render player
+     *
+     * @private
+     */
+	_renderPlayer() {
+		if (!this._player) {
+			// Add loading class
+			this.$el.addClass(this.options.classLoading);
+
+			// Use ApiLoader service
+			this.options.loader.requestPlayer().done(this._onPlayerReceived);
+		}
+	}
+
+	/**
+	 * Update interval
+	 *
+	 * @private
+	 */
+	_updateInterval() {
+		if (!this._interval) {
+			this._interval = window.setInterval(
+				this._onInterval,
+				this.options.playerProgressInterval
+			);
+			this._debug('updateInterval');
+		}
+	}
+
+	/**
+	 * Reset interval
+	 *
+	 * @private
+	 */
+	_resetInterval() {
+		if (this._interval) {
+			window.clearInterval(this._interval);
+			this._interval = undefined;
+			delete(this._interval);
+			this._debug('resetInterval');
+		}
+	}
+
+	/**
+	 * Update video progress
+	 *
+	 * @private
+	 */
+	_updateProgress() {
+		if (this._hasPlayer()) {
+			var self = this,
+				steps = self.options.playerProgressSteps;
+
+			self._player.getCurrentTime().then(function(seconds) {
+				self._player.getDuration().then(function(duration) {
+					var progress = seconds / duration * 100;
+
+					progress = Math.floor(progress / steps) * steps;
+					progress = Math.max(self._progress, progress);
+
+					if (progress !== self._progress) {
+						self._progress = progress;
+						self.context.dispatch('vimeoplayer:updateprogress', self);
+						self._debug('updateProgress', self._progress);
+					}
+				});
+			});
+		}
+	}
+
+	/**
+	 * Reset video progress
+	 *
+	 * @private
+	 */
+	_resetProgress() {
+		this._progress = -1;
+		this._debug('resetProgress', this._progress);
+	}
+
+	//================================================================================
+	// Events
+	//================================================================================
+
+	/**
+     * Bind events
+     *
+     * @private
+     */
+	_bindEvents() {
+		_.bindAll(
+			this,
+			'_onClickPlay',
+			'_onPlayerReceived',
+			'_onInterval',
+			'_onPlay',
+			'_onPause',
+			'_onEnded',
+			'_onLoaded',
+			'_onError',
+			'_onReady'
+		);
+
+		this.$el.on('click', this.options.trigger, this._onClickPlay);
+	}
+
+	/**
+	 * Play event handler
+	 *
+	 * @private
+	 */
+	_onPlayHandler() {
+		this._updateProgress();
+		this._updateInterval();
+		this.playMedia();
+		this.showDisplay();
+		this.context.dispatch('vimeoplayer:play', this);
+	}
+
+	/**
+	 * Stop event handler
+	 *
+	 * @private
+	 * @param {string} [eventName] The event name to dispatch onto the Context's Event Bus
```

Comment:
Why using `[eventName]` in brackets? Means this optional? Document the default value...